contracts/StrategyRegistry.vy

```diff
@@ -0,0 +1,70 @@
+# @version 0.2.12
+
+governance: public(address)
+pendingGovernance: public(address)
+
+event NewGovernance:
+    governance: address
+
+interface Strategy:
+    def name() -> String[64]: view
+
+struct StrategyInfo:
+    name: String[64]
+    numReleases: uint256 
+    versions: address[20]
+
+strategiesReleases: public(HashMap[bytes32, StrategyInfo])
+
+@external
+def __init__():
+    self.governance = msg.sender
+
+@external
+def setGovernance(governance: address):
+    """
+    @notice Starts the 1st phase of the governance transfer.
+    @dev Throws if the caller is not current governance.
+    @param governance The next governance address
+    """
+    assert msg.sender == self.governance  # dev: unauthorized
+    self.pendingGovernance = governance
+
+
+@external
+def acceptGovernance():
+    """
+    @notice Completes the 2nd phase of the governance transfer.
+    @dev
+        Throws if the caller is not the pending caller.
+        Emits a `NewGovernance` event.
+    """
+    assert msg.sender == self.pendingGovernance  # dev: unauthorized
+    self.governance = msg.sender
+    log NewGovernance(msg.sender)
+
+@external
+def addNewRelease(strategy :address, name: String[64] = ""):
+    """
+    @notice add a new version of a strategy
+    """
+    assert msg.sender == self.pendingGovernance  # dev: unauthorized
+    strategyName: String[64] = name 
+    if name == "":
+        strategyName = Strategy(strategy).name()
+    strategyHash: bytes32 = sha256(strategyName)
+    if self.strategiesReleases[strategyHash].numReleases == 0:
+	    self.strategiesReleases[strategyHash] = StrategyInfo({
+		    name: strategyName,
+		    numReleases: 0,
+		    versions: [ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS, ZERO_ADDRESS]
+	    })
+    numReleases: uint256 = self.strategiesReleases[strategyHash].numReleases
+    self.strategiesReleases[strategyHash].numReleases = numReleases + 1
+    self.strategiesReleases[strategyHash].versions[numReleases] = strategy
```

Comment:
what's the usecase for tracking multiple releases of the same strategy name?

Reply:
So I tough that should work like the registry and keep multiple versions for the same name.
What we need is to handle multiple vault versions for the same strategy name, this can be handled by adding the vault version to the name of the strategy or we save it as part of the hashmap

Reply:
Don't forget that some of this information might already exist in the Vault registry

Reply:
Can't do that because Dictionary keys must be a base type.

Reply:
ah well, `base32` should work.

perhaps should add the ability for any type that can be hashed to be used as a key, since that's really the requirement here.